microsoft/testsuites/kata_conformance/kata_conformance_tests.py

```diff
@@ -0,0 +1,117 @@
+# Copyright (c) Microsoft Corporation.
+# Licensed under the MIT license.
+import json
+from pathlib import Path
+from typing import Any, Dict
+
+from lisa import (
+    Environment,
+    Logger,
+    Node,
+    TestCaseMetadata,
+    TestSuite,
+    TestSuiteMetadata,
+)
+from lisa.operating_system import CBLMariner, Ubuntu
+from lisa.testsuite import TestResult
+from lisa.tools import Lscpu
+from lisa.util import LisaException, SkippedException
+from microsoft.testsuites.kata_conformance.aks_infra import AKSInfra
+from microsoft.testsuites.kata_conformance.kata_conformance_tests_tool import (
+    KataConformanceTests,
+)
+
+
+def validate_variable(variables, key, log, nullable=False):
+    log.debug(f"Checking for param : {key} in variables , nullable : {nullable}")
+    value = variables.get(key, None)
+    if (not nullable and value is None) or (
+        not nullable and type(value) == str and value.strip() == ""
+    ):
+        raise LisaException(key + " : Variable is null/empty")
+    else:
+        return value
+
+
+@TestSuiteMetadata(
+    area="kubernetes",
+    category="community",
+    description="""
+    This test suite is for executing the conformance test with kata container
+    which uses infra of azure AKS as kubernetes cluster
+    """,
+)
+class KataConformanceTestSuite(TestSuite):
+    @TestCaseMetadata(
+        description="""
+            Runs Kubernetes Kata Cotaniner Conformance test.
+        """,
+        priority=3,
+        timeout=36000,
```

Comment:
It means 10 hours? How long the test will take actually? If it's 10 hours, move it to P4.

Reply:
For slower, system it takes around 8h. I changed it to 5h as of now. Made it P4

Reply:
Is it possible to reduce test time? It's ok to keep 10 hours, but as a normal test condition, it takes too long to complete if it takes more than 1 hour. Please check if it's possible to reduce running time.

Reply:
Kata test takes more than 1 hour as it includes AKS infra creation, Kubernetes Build, Kata webhook deployment on created AKS and then it launches the Conformance test which are 350+ in number.

Reply:
Do you mean the whole progress is around 1 hour in normal situation? If so, it can be P3. If it's more than 2 hours, it should be moved to P4.

Reply:
Yes it can take more than2 Hours. It depends on VM size from where we are launching the test. In this case, it would be the agent vm where pipeline is running. It will create AKS infra on Azure and try to run conformance test there but it can be launched from anywhere.

Reply:
Will make this P4